components/textbox/textbox.tsx

```diff
@@ -77,7 +77,17 @@ export default class Textbox extends React.PureComponent<Props> {
     constructor(props: Props) {
         super(props);
 
-        this.suggestionProviders = [
+        this.suggestionProviders = [];
+
+        if (props.supportsCommands) {
+            this.suggestionProviders.push(new CommandProvider({
+                teamId: this.props.currentTeamId,
+                channelId: this.props.channelId,
+                rootId: this.props.rootId,
+            }));
+        }
```

Comment:
If I understand correctly, by putting this provider first in the list, we are giving the `CommandProvider` precedence over all other providers here. This allows the provider to avoid being "stomped on" by the other providers.

One thing that comes to mind is non-app slash commands. Do the user and channel providers still work as intended with this change? Something like `/todo add ask @daniel what he thinks about this`

Reply:
You are right. Non-app slash commands will loose the ability to add mentions. Also, mentions inside text will be not possible. (e.g. `/todo add --message "Ask @daniel what he thinks about this"`).

Reply:
That sounds like something we should figure out. I don't think we can make the other commands work worse in favour of how Apps handle at mentions.

Reply:
In order for the other commands to work the same is to do something I do not like very much:
- Let the app parser take preference, but do not handle the base command (only after the base command is in place, the app parser takes over the autocomplete)
- Maintain the command provider as a separate provider of the app parser. Nevertheless, it will still consider the base command, so all command suggestions are in the same place.